insights/client/__init__.py

```diff
@@ -48,7 +48,9 @@ def __init__(self, config=None, setup_logging=True, **kwargs):
         if setup_logging:
             self.set_up_logging()
             try_auto_configuration(self.config)
-
+        else:
+            # write PID to file in case we need to ping systemd
+            write_to_disk(constants.pidfile, content=str(os.getpid()))
```

Comment:
This will only write the pidfile on the very first initialization of `InsightsClient` from the `/usr/bin/insights-client` init code (when `setup_logging=False`, before phases are run. So if a phase were to die and roll to the next egg, the pidfile would be untouched as the main `insights-client` process is still running.